src/lib.rs

```diff
@@ -893,6 +930,57 @@ unsafe impl<T> KnownLayout for [T] {
         //   struct `Foo(i32, [u8])` or `(u64, Foo)`.
         slc.len()
     }
+
+    #[cfg(feature = "alloc")]
+    unsafe fn drop(slf: &mut UnalignUnsized<Self>) {
```

Comment:
Does this play nicely with `Pin`'d types?

Reply:
Yes, Pin bars you from moving the referent before running its destructor. It doesn't place any restrictions on what you do in the destructor.